NumberTheory/BinomialCoefficient.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+    Binomial Coefficient:
+    For a set containing N unique elements, find the number of subsets containing K elements (aka N choose K)
+    This approach uses the recurrence BinomialCoefficient(N,K) = BinomialCoefficient(N-1,K) + BinomialCoefficient(N-1,K-1),
+    and a dynamic programming approach to memoize (save) BinomialCoefficient values computed between calls
+
+    Time complexity:
+    O(N*K) - cache[i][j] will be computed at most once for each i<=N and j<=K
+
+    Space complexity:
+    O(N*K) due to memoization (due to the 'cache' variable), and O(N) stack space (due to recursion)
+*/
+
+#include <iostream>
+#include <vector>
+using namespace std;
+
+const int FLAG = -1;
+
+int BinomialCoefficientHelper(int N, int K, vector<vector<int> > cache)//uses recurrence relation and fills cache to find binomial coefficient of N & K
```

Comment:
Pass `cache` by reference to prevent creating copies of it during each recursive call.